datasource-remote/src/main/java/cmm/apps/esmorga/datasource_remote/api/authenticator/EsmorgaAuthInterceptor.kt

```diff
@@ -0,0 +1,20 @@
+package cmm.apps.esmorga.datasource_remote.api.authenticator
+
+import android.content.SharedPreferences
+import cmm.apps.esmorga.datasource_remote.api.authenticator.AuthorizationConstants.AUTHORIZATION_HEADER_KEY
+import cmm.apps.esmorga.datasource_remote.api.authenticator.AuthorizationConstants.AUTHORIZATION_HEADER_VALUE
+import cmm.apps.esmorga.datasource_remote.api.authenticator.AuthorizationConstants.SHARED_AUTH_TOKEN_KEY
+import okhttp3.Interceptor
+import okhttp3.Response
+import org.koin.core.component.KoinComponent
+
+class EsmorgaAuthInterceptor(private val sharedPreferences: SharedPreferences) : Interceptor, KoinComponent {
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val requestBuilder = chain.request().newBuilder()
```

Comment:
this will add the Auth header to all calls, right? Would it be possible to mark them with an annotation or similar so it is only added to the ones we decide?